collections/_drafts/agent-forwarding.md

```diff
@@ -0,0 +1,275 @@
+other token. It also supports Secure Shell (SSH) by implementing the ssh\-agent
+protocol.
+
+The traditional way to run _gpg\_\_\-agent_ on Unix systems is by launching it
+at login time and using an environment variable (GPG_AGENT_INFO) to tell the
+other GnuPG modules how to connect to the agent. However, correctly managing the
+startup and this environment variable is cumbersome, so a more straightforward
+method is required. Since GnuPG 2.0.16, the \--use-standard-socket option
+already allowed starting the agent on the fly; however, the environment variable
+was still needed.
+
+With GnuPG 2.1, the need for GPG_AGENT_INFO has been completely removed, and the
+variable is ignored.
+
+Instead, a fixed Unix domain socket named S.gpg\-agent in the GnuPG home
+directory (by default ~/.gnupg) is used. The agent is also started on-demand by
+all tools requiring services from the agent.
+
+If the option \--enable-ssh\-support is used, the auto-start mechanism does not
+work because _ssh_ does not know about this mechanism. Instead, the environment
+variable SSH_AUTH_SOCK must be set to the S.gpg-agent.ssh socket in the GnuPG
+home directory. Further, _gpg\_\_\-agent_ must be started by either using a
+GnuPG command that implicitly starts gpg\-agent or by using \`gpgconf --launch
+gpg\-agent\` to explicitly start it without first having to use a GnuPG command.
+
+gpg\-agent is a daemon to manage secret (private) keys independently from any
+protocol. It is a backend for gpg, gpgsm, and other utilities.
+
+GPG Agent Configuration
+
+There are a few configuration files needed for the operation of the agent. They
+may all be found in the current GnuPG home directory, which defaults
+to ~/.gnupg.
+
+\[\[windows\]\]
+
+However, there may be problems on Windows systems with Gpg4Win installed;
+**Gpg4Win may have changed this default GnuPG home directory location** to an
+AppData subdirectory (i.e., C:\\Users\\<username>\\AppData\\Roaming\\gnupg).
+This location, however, is not the location our GPG agent will be using, so to
+reaffirm our preference for the default location, we will set
+the GNUPGHOME environment variable to ~/.gnupg in the Git Bash startup script by
+running the following.
+
+echo 'export GNUPGHOME="~/.gnupg"' >> .bashrc
+
+\[\[endwindows\]\]
+
+Setting environment variables
+
+Add the following lines to your .bashrc or whatever initialization file is used
+for all shell invocations:
+
+GPG_TTY=$(tty)
+
+export GPG_TTY
+
+This variable may only be helpful if you use \`pinentry\-curses\` (the
+terminal-based pin entry program).
+
+\[\[windows\]\]
+
+On Windows systems, you should add the above lines to the ~/.bashrc file for use
+by Git Bash.
+
+\[\[endwindows\]\]
+
+GnuPG configuration
+
+It's important to note that GnuPG on the remote system still needs your public
+GPG keys to work correctly. So you have to ensure they are available on the
+remote system even if your secret keys are not.
+
+\[\[note\]\]
```

Comment:
```suggestion
\[\[note\]\]
```

btw, this cute lil admonition syntax was previously ruled as being non-preferable

Reply:
<blockquote class="twitter-tweet"><p lang="en" dir="ltr">I personally have the same feedback as before: should&#39;ve been a directive.<br><br>Square brackets are used a lot already in markdown. The things they came up with conflict with a link definition ([!note]: &lt;&gt;)</p>&mdash; Titus 🇵🇸 (@wooorm) <a href="https://twitter.com/wooorm/status/1701865524236345734?ref_src=twsrc%5Etfw">September 13, 2023</a></blockquote>

Reply:
this would be a nice time for windings font selection

we'd have some diverse surrounding content and not all the text is uniform

🔗 https://dereknongeneric.github.io/agent-forwarding-eye-candy/agent-forwarding.html

Reply:
y'know, at first i was having trouble deciding
whether to keep the leading section numbers or
move that to some other place that doesn't interfere,
but now i really like how their addition makes it look academic

Reply:
btw, the HTML to Markdown conversion was done via online tool:

🔗 https://htmltomd.com/

briefly glancing at the output; it had stripped wayyy more tags than Pandoc
so, i was like "screw it, nobody else even comes close", but not sure why
all the backslashes have been applied to basically anything that can be

Reply:
re-reading https://github.com/OpenINF/GitHub-Markdown/issues/4 now and
gearing up for its demolition, which will occur in ~30 min (08:00pm EST)

who like me still feels weird about using the `aside` HTML tag for them?

- http://html5doctor.com/understanding-aside/
- https://developers.google.com/style/notices
- https://docs.anaconda.com/restructuredtext/index.html#admonitions-notes

maybe i will feel better about it after reading those things on them

Reply:
last note worth mentioning in-line, below stack of links to style guide convetions else where; we used to
have our own freaking baby styleguide in our top-level
org dotrepo:

🔗 https://github.com/OpenINF/.github/commit/f720bce548a0a2a84c8a95f9f8a62528eb1b873f

in retrospect, i do not change my opinion about that being the wrong place for that handbook sub-project, which i was imagining would outgrow (in amount of content) the repo it had started in and the dot-prefix-namesake would soon have made no sense at all

regretting my decision of not migrating it else where tho

Reply:
```suggestion
:::note{.note}
```

Reply:
> the actual resulting suggestion is better than this (gimme a sec to dig it up)

🔗 https://github.com/micromark/micromark-extension-directive?tab=readme-ov-file#syntax

`micromark-extension-directive` has definitely thought this out more clearly